tests/test_pay.py

```diff
@@ -6702,3 +6702,19 @@ def get_local_channel_by_id(node, chanid):
     receipt = only_one(l3.rpc.listinvoices("inv")["invoices"])
     assert receipt["status"] == "paid"
     assert receipt["amount_received_msat"] == total_msat
+
+
+def test_pay_unannounced_routehint(node_factory, bitcoind):
+    """Tests whether sender can pay recipient through unannounced channels with
+    2 hops where the second hop uses a route hint."""
+    l1, l2, l3, r1 = node_factory.get_nodes(4)
+
+    # l1 and l3 are connected with unannounced channels to l2
+    node_factory.join_nodes([l1, l2], announce_channels=False)
+    node_factory.join_nodes([l2, l3], announce_channels=False)
+
+    # Need this to ensure l3 adds a route hint in the invoice
+    node_factory.join_nodes([l2, r1], wait_for_announce=True)
+
+    invoice = l3.rpc.invoice(2100000, 'receive', 'receive')
+    l1.rpc.pay(invoice['bolt11'])
```

Comment:
I would preferer to just not blindly trust this by avoiding to check the result.

Reply:
Updated